### DISCUSSION TITLE
Missing/Incorrect Program Outline

### DISCUSSION THREAD

**Discussion by JonFP** [codefori/vscode-ibmi#609] [Q&A]
For a while now I have found that some programs just do not populate the outline even if (as far as the compiler is concerned) they are syntactically correct.  Yesterday I had one program that had an outline but the outline did not change when I deleted/changed lines (and I deleted a lot). In fact the outline did not update even after I closed and re-opened the outline view nor when I closed the file and then re-opened it. It was still the same outdated outline. Only after closing the file, disconnecting, and re-opening the file did I get an updated outline.  I can't find a refresh option for the outline so I can't force it to refresh.

So two questions really:
1) Is there any way to find out why no outline was built for a valid program.
2) Is there a way to refresh outline somehow.

**Comment by worksofliam** [comment]
> Is there any way to find out why no outline was built for a valid program.

If there is no outline view, check the RPGLE language tools output channel. There might be an error in there which you can share here.

As for the refresh, there is no way to force it, but I may need to look deeper into this.

**Comment by JonFP** [comment]
Here are the files requested.  YAJL is in the library list and the compile is fine. I guess because the outline failed the /include does not hyperlink
[ForLiamLog.txt](https://github.com/halcyon-tech/vscode-ibmi/files/8411394/ForLiamLog.txt)
[ForLiamYAJLPLAY.RPGLE.txt](https://github.com/halcyon-tech/vscode-ibmi/files/8411399/ForLiamYAJLPLAY.RPGLE.txt)
.
